src/mlpack/core/data/dataset_info.hpp

```diff
@@ -54,7 +59,8 @@ class DatasetInfo
    * @param string String to find/create mapping for.
    * @param dimension Index of the dimension of the string.
    */
-  size_t MapString(const std::string& string, const size_t dimension);
+  typename MapPolicy::map_type_t MapString(const std::string& string,
+                                            const size_t dimension);
```

Comment:
I think we can make map_type_t as mapped_type, mapped_type is same as the typedef of std::map, mapped_type should be more familiar with c++ programmer(my example did not name it well, sorry about that)